client/my-sites/patterns/components/pattern-preview/index.tsx

```diff
@@ -79,6 +80,10 @@ function PatternPreviewFragment( {
 	useTimeoutToResetBoolean( isPermalinkCopied, setIsPermalinkCopied );
 	useTimeoutToResetBoolean( isPatternCopied, setIsPatternCopied );
 
+	useEffect( () => {
+		ref.current?.dispatchEvent( new CustomEvent( 'patternPreviewResize', { bubbles: true } ) );
```

Comment:
Every pattern preview has its own resize observer, and will dispatch the custom event. At the same time, from what I understand, we don't really care about each pattern preview resizing; we care about resizing the entire pattern gallery container, don't we? Wouldn't it be more optimal to observe that container for resizing instead, and go through all items to set the `marginTop`? This way we won't need a custom event, either.

Reply:
We do care about changes to the pattern preview sizes. They will resize while loading, and this is a simple way to capture those changes.